internals/search.py

```diff
@@ -233,30 +238,22 @@ def process_query(
   # 3. Get the result of each future and combine them into a result ID set.
   logging.info('now waiting on futures')
 
-  # 3a. Process all negation operations.
+  # 3a. Process user query: negation, AND, and OR.
   feature_id_future_ops = process_negation_operations(feature_id_future_ops)
+  query_clauses = process_and_operations(feature_id_future_ops)
+  result_id_set = process_or_operations(query_clauses)
 
-  # 3b. Process all AND operations.
-  or_clauses = process_and_operations(feature_id_future_ops)
-
-  # 3c. Process all OR operations.
-  result_id_set = process_or_operations(or_clauses)
-
-  # 3d. Process all permission ops.
-  permission_ids = process_and_operations(permissions_future_ops)
-  if len(permission_ids) == 1:
-    if terms == []:
-      # Empty search terms.
-      result_id_set = permission_ids[0]
-    else:
-      result_id_set.intersection_update(permission_ids[0])
+  # 3b. Process all permission ops, then interesect to apply permisisons.
+  permission_clauses = process_and_operations(permissions_future_ops)
+  permission_ids = process_or_operations(permission_clauses)
```

Comment:
In the case of permission clauses, OR clauses only apply within permission terms, rather than the whole query?

Reply:
Yes, I suppose that is true:  If there was ever an 'OR' in the permissions clauses that 'OR' would apply only to the permissions clauses. I think that makes sense, however I don't foresee that we would use 'OR'.  The main reason to make this call here is to convert from clauses to IDs in the same way that is done for the user search terms.